server/src/models/productModel.ts

```diff
@@ -0,0 +1,27 @@
+import { Schema, model, Types } from "mongoose";
+
+interface Exercise {
+  _id: Types.ObjectId;
+  name: string;
+  sets?: number;
+  reps?: number;
+}
+
+interface ExerciseSet {
+  _id: Types.ObjectId;
+  date: Date;
+  exercises: Types.Array<Exercise>;
+}
+
+const setSchema = new Schema<Exercise>({
+  name: { type: String, required: true },
+  sets: { type: Number, required: true },
```

Comment:
This required leads me to believe you want sets and reps everywhere 🤔

Reply:
resolved in comment above